### PR TITLE
add companies inside entity page

### PR DESCRIPTION
**Pull request by carlorusso**

### CODE REVIEW

**Comment by tobymao** [comment]
what does this look like

@jenf could you review this?

**Comment by carlorusso** [comment]
> what does this look like
> 
> @jenf could you review this?

On desktop it's like this
image

**Comment by jenf** [comment]
Is this trying to show privates that are owned by the bank?

**Comment by carlorusso** [comment]
> Is this trying to show privates that are owned by the bank?

no, only the privates without any owner

File: assets/app/view/game/entities.rb [hunk 1]
```diff
@@ -47,13 +47,19 @@ def render
           h(Bank, game: @game),
           h(GameInfo, game: @game, layout: 'upcoming_trains'),
           *@game.corporations.select(&:receivership?).map { |c| h(Corporation, corporation: c) },
+          render_unsold_companies,
```
Comment (jenf) [inline assets/app/view/game/entities.rb:50]: Given that 1860 should probably show the unowned company if the bank owns this 'unsold' is incorrect and probably should be generalized.

render_unowned_companies would be better

File: assets/app/view/game/un_owned_companies.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require 'view/game/company'
+
+module View
+  module Game
+    class UnOwnedCompanies < Snabberb::Component
```
Comment (jenf) [inline assets/app/view/game/un_owned_companies.rb:7]: UnownedCompanies

File: lib/engine/game/base.rb [hunk 3]
```diff
@@ -1278,6 +1278,11 @@ def company_sellable(company)
         !company.owner.is_a?(Corporation)
       end
 
+      # Called by View::Game::Entities to determine if the company should be shown on entities
+      def unsold_purchasable_companies(_company)
```
Comment (jenf) [inline lib/engine/game/base.rb:1282]: unowned

**Comment by jenf** [comment]
In addition, please see if you can sort the formatting so that it is fixed width ala the player & corporation cards

File: assets/app/view/game/entities.rb (hunk 1 above)
Comment (carlorusso) [reply assets/app/view/game/entities.rb]: done

File: assets/app/view/game/un_owned_companies.rb (hunk 2 above)
Comment (carlorusso) [reply assets/app/view/game/un_owned_companies.rb]: done

File: lib/engine/game/base.rb (hunk 3 above)
Comment (carlorusso) [reply lib/engine/game/base.rb]: done

**Comment by carlorusso** [comment]
> In addition, please see if you can sort the formatting so that it is fixed width ala the player & corporation cards

is ok now?

File: lib/engine/game/base.rb [hunk 4]
```diff
@@ -1278,6 +1278,11 @@ def company_sellable(company)
         !company.owner.is_a?(Corporation)
       end
 
+      # Called by View::Game::Entities to determine if the company should be shown on entities
+      def unowned_purchasable_companies(_company)
```
Comment (ryandriskel) [inline lib/engine/game/base.rb:1282]: Current Entity usually isn't a company. Should this be _entity ?

File: assets/app/view/game/entities.rb [hunk 5]
```diff
@@ -47,13 +47,19 @@ def render
...
           *extra_bank,
         ].compact)
 
         children.concat(bankrupt_players.map { |p| h(:div, [h(Player, player: p, game: @game)]) })
 
         h('div#entities', div_props, children)
       end
+
+      def unowned_companies
+        companies = @game.unowned_purchasable_companies(@current_entity)
```
Comment (ryandriskel) [inline assets/app/view/game/entities.rb:60]: Is @current_entity defined in this context?

**Comment by ryandriskel** [comment]
18CO could benefit from some way to show this information as privates not bought during the auction can later be bought from the bank by corporations, but the only place to find out which ones are available is in the hidden "Show Companies of Other Players" area during an OR.

**Comment by carlorusso** [comment]
> 18CO could benefit from some way to show this information as privates not bought during the auction can later be bought from the bank by corporations, but the only place to find out which ones are available is in the hidden "Show Companies of Other Players" area during an OR.

Yes, I’m doing this feature for 18ZOO, but it should be useful for other title too

**Comment by jenf** [comment]
Please update picture if the UI has been changed to fix the width

**Comment by carlorusso** [comment]
> Please update picture if the UI has been changed to fix the width

I'm waiting https://github.com/tobymao/18xx/pull/3494 to update this too

File: assets/app/view/game/entities.rb (hunk 5 above)
Comment (carlorusso) [reply assets/app/view/game/entities.rb:63]: yes, it's a player or a corporation

**Comment by carlorusso** [comment]
@jenf updated the image too; maybe the UI can improved, but I don't know how to do, so if you are ok I should prefer to go with this for now

**Pull request merged**